lib/kernel-manager.coffee

```diff
@@ -98,7 +74,13 @@ class KernelManager
 
 
     getAllKernelSpecs: ->
-        return _.map @_kernelSpecs, 'spec'
+        # TODO(nikita): switch to async and jupyter-js-services API
+        # Note that using sync here is a temporary regression, because it forces
+        # a manual call to update-kernels by the user.
+        # This will be fixed in a later patch, that adds proper async back
+        kernelSpecs = zmq_services.getZMQGateway().getKernelSpecsSync()
+        _.forEach kernelSpecs, (obj, name) -> obj.spec.name = name
```

Comment:
Hydrogen uses `display_name`s that are part of the kernel spec (see [here](http://jupyter-client.readthedocs.io/en/latest/api/kernelspec.html)).

We've gone through a lot of trouble to minimise the names used in Hydrogen. Currently, we use only two:
- lowercased grammar languages (to determine what kernels can run a file)
- display_names (this is the name shown to the user to identify the kernel)

Can we make do without using that kernel name?